tools/image_processing/imagej2/imagej2_analyze_skeleton/imagej2_analyze_skeleton.xml

```diff
@@ -78,18 +78,28 @@
         </test>
     </tests>
     <help>
+
+.. class:: warningmark
+
+The underlying AnalyzeSkeleton Fiji plugin works on binary images, so other image types will
+automatically be converted to binary before they are analyzed.  This step is performed using the
+ImageJ2 **Make Binary** command with the following settings: **Iterations:** 1, **Count:** 1,
+**Black background:** No, **Pad edges when eroding:** No.  If these settings are not appropriate,
+first manually convert the image to binary using the **Convert to binary (black and white) with
```

Comment:
Hi Bjorn,

Are you still in Baltimore?  I hope your visit is going well!



I have my docker-galaxy-csg image built on docker-hub:

https://hub.docker.com/r/gregvonkuster/docker-galaxy-csg/builds/ <https://hub.docker.com/r/gregvonkuster/docker-galaxy-csg/builds/>


My Dockerfile includes this - I have to use the dev tag from your docker-galaxy-stable image because 15.07 is missing some files I need.  

FROM bgruening/galaxy-stable:dev


So I can now do this:

$ docker pull gregvonkuster/docker-galaxy-csg:master
master: Pulling from gregvonkuster/docker-galaxy-csg

8b1884ea1bde: Pulling fs layer 
aacc7221ffdc: Pulling fs layer 
aa0a5fcc3686: Pulling fs laye
……





When the container is built, I have something like this:

$ docker images
REPOSITORY                        TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
gregvonkuster/docker-galaxy-csg   master              991f5da05279        About an hour ago   2.671 GB




I can start the container:

$ docker run -i -t -p 8080:80 991f5da05279
umount: /var/lib/docker: must be superuser to umount
tmpfs on /proc/kcore type tmpfs (rw,nosuid,mode=755)
Disable Galaxy Interactive Environments. Start with --privileged to enable IE's.
/usr/lib/python2.7/dist-packages/supervisor/options.py:295: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
  'Supervisord is running as root and it is searching '
Starting ProFTP
proftpd: started
Starting Galaxy reports webapp
reports: started
Starting nodejs
galaxy:galaxy_nodejs_proxy: ERROR (abnormal termination)
==> /home/galaxy/handler0.log <==
galaxy.jobs.runners.drmaa INFO 2015-11-21 17:51:09,477 Overriding DRMAA_LIBRARY_PATH due to runner plugin parameter: /usr/lib/slurm-drmaa/lib/libdrmaa.so
galaxy.jobs.runners DEBUG 2015-11-21 17:51:09,496 Starting 2 SlurmRunner workers
galaxy.jobs DEBUG 2015-11-21 17:51:09,497 Loaded job runner 'galaxy.jobs.runners.slurm:SlurmJobRunner' as 'slurm'
galaxy.jobs.handler DEBUG 2015-11-21 17:51:09,497 Loaded job runners plugins: slurm
galaxy.jobs.handler INFO 2015-11-21 17:51:09,501 job handler stop queue started
galaxy.jobs.handler INFO 2015-11-21 17:51:09,523 job handler queue started
galaxy.sample_tracking.external_service_types DEBUG 2015-11-21 17:51:09,525 Loaded external_service_type: Simple unknown sequencer 1.0.0
galaxy.sample_tracking.external_service_types DEBUG 2015-11-21 17:51:09,530 Loaded external_service_type: Applied Biosystems SOLiD 1.0.0
galaxy.workflow.scheduling_manager DEBUG 2015-11-21 17:51:09,547 Starting workflow schedulers
galaxy.queue_worker INFO 2015-11-21 17:51:09,562 Binding and starting galaxy control worker for handler0

==> /home/galaxy/handler1.log <==
galaxy.jobs.runners.drmaa INFO 2015-11-21 17:51:09,478 Overriding DRMAA_LIBRARY_PATH due to runner plugin parameter: /usr/lib/slurm-drmaa/lib/libdrmaa.so
galaxy.jobs.runners DEBUG 2015-11-21 17:51:09,496 Starting 2 SlurmRunner workers
galaxy.jobs DEBUG 2015-11-21 17:51:09,496 Loaded job runner 'galaxy.jobs.runners.slurm:SlurmJobRunner' as 'slurm'
galaxy.jobs.handler DEBUG 2015-11-21 17:51:09,496 Loaded job runners plugins: slurm
galaxy.jobs.handler INFO 2015-11-21 17:51:09,497 job handler stop queue started
galaxy.jobs.handler INFO 2015-11-21 17:51:09,541 job handler queue started
galaxy.sample_tracking.external_service_types DEBUG 2015-11-21 17:51:09,542 Loaded external_service_type: Simple unknown sequencer 1.0.0
galaxy.sample_tracking.external_service_types DEBUG 2015-11-21 17:51:09,543 Loaded external_service_type: Applied Biosystems SOLiD 1.0.0
galaxy.workflow.scheduling_manager DEBUG 2015-11-21 17:51:09,560 Starting workflow schedulers
galaxy.queue_worker INFO 2015-11-21 17:51:09,575 Binding and starting galaxy control worker for handler1

==> /home/galaxy/reports.log <==
galaxy.web.framework.base DEBUG 2015-11-21 17:51:07,887 Enabling 'sample_tracking' controller, class: SampleTracking
galaxy.web.framework.base DEBUG 2015-11-21 17:51:07,897 Enabling 'users' controller, class: Users
galaxy.web.framework.base DEBUG 2015-11-21 17:51:07,898 Enabling 'home' controller, class: HomePage
galaxy.webapps.reports.buildapp DEBUG 2015-11-21 17:51:07,899 Enabling 'httpexceptions' middleware
galaxy.webapps.reports.buildapp DEBUG 2015-11-21 17:51:07,907 Enabling 'recursive' middleware
galaxy.webapps.reports.buildapp DEBUG 2015-11-21 17:51:07,927 Enabling 'error' middleware
galaxy.webapps.reports.buildapp DEBUG 2015-11-21 17:51:07,927 Enabling 'trans logger' middleware
galaxy.webapps.reports.buildapp DEBUG 2015-11-21 17:51:07,936 Enabling 'x-forwarded-host' middleware
Starting server in PID 81.
serving on http://127.0.0.1:9001 <http://127.0.0.1:9001/>

==> /home/galaxy/uwsgi.log <==
galaxy.webapps.galaxy.buildapp DEBUG 2015-11-21 17:51:10,174 Enabling 'x-forwarded-host' middleware
galaxy.webapps.galaxy.buildapp DEBUG 2015-11-21 17:51:10,175 Enabling 'Request ID' middleware
Sat Nov 21 17:51:10 2015 - WSGI app 0 (mountpoint='') ready in 13 seconds on interpreter 0xc32f60 pid: 42 (default app)
Sat Nov 21 17:51:10 2015 - *** uWSGI is running in multiple interpreter mode ***
Sat Nov 21 17:51:10 2015 - spawned uWSGI master process (pid: 42)
Sat Nov 21 17:51:10 2015 - spawned uWSGI worker 1 (pid: 118, cores: 4)
Sat Nov 21 17:51:10 2015 - spawned uWSGI worker 2 (pid: 119, cores: 4)
Sat Nov 21 17:51:10 2015 - *** Stats server enabled on 127.0.0.1:9191 fd: 11 ***
galaxy.queue_worker INFO 2015-11-21 17:51:10,183 Binding and starting galaxy control worker for main.1
galaxy.queue_worker INFO 2015-11-21 17:51:10,187 Binding and starting galaxy control worker for main.2


I can then access the Galaxy-csg flavor in my browser:




However, when I try to execute a job, I always seem to get this following in the paster log, and the job cannot run.  Have you seen this before?

==> /home/galaxy/handler1.log <==
galaxy.jobs.runners.drmaa WARNING 2015-11-21 17:52:30,797 (1) drmaa.Session.runJob() failed, will retry: code 1: slurm_submit_batch_job: Unable to contact slurm controller (connect failure)